src/commands.cpp

```diff
@@ -8261,4 +8261,114 @@ bool CommandBackupRemove::procresult(Result r)
     return r.wasErrorOrOK();
 }
 
+CommandGetBanners::CommandGetBanners(MegaClient* client)
+{
+    cmd("gban");
+
+    tag = client->reqtag;
+}
+
+bool CommandGetBanners::procresult(Result r)
+{
+    if (r.wasErrorOrOK())
+    {
+        client->app->getbanner_result(r.errorOrOK());
+        return true; // because parsing didn't fail
+    }
+
+    /*
+    [
+        [
+            {
+                "id": 2, ///The banner id
+                "t": "R2V0IFZlcmlmaWVk", ///Banner title
+                "d": "TWFrZSBpdCBlYXNpZXIgZm9yIHlvdXIgY29udGFjdHMgdG8gZmluZCB5b3Ugb24gTUVHQS4", ///Banner description.
+                "img": "Verified_image.png", ///Image name.
+                "l": "", ///URL
+                "bimg": "Verified_BG.png", ///background image name.
+                "dsp": "https://web-sandbox3.developers.mega.co.nz/images/mega/" ///Where to get the image.
+            },
+            {
+                ...
+            }
+        ]
+    ]
+    */
+
+    if (!client->json.enterarray() ||
+        !client->json.enterarray())
+    {
+        client->app->getbanner_result(API_EARGS);
+        return false; // parsing failed
+    }
+
+    vector< tuple<int, string, string, string, string, string, string> > banners;
+
+    // loop array elements
+    while (client->json.enterobject())
+    {
+        int fieldcount = 7,
+            id = 0;
```

Comment:
We try to avoid declaration of multiple variables in a row like this ones, with initialization.
Also, use cammelCase for naming variables, please.

It's preferred to do:
```suggestion
        int fieldCount = 7;
        int id = 0;
```